tomviz/CentralWidget.cxx

```diff
@@ -325,23 +351,15 @@ void CentralWidget::setDataSource(DataSource* source)
     }
 
   // Calculate a histogram.
-  vtkNew<vtkTable> table;
+  vtkSmartPointer<vtkTable> table =
+    vtkSmartPointer<vtkTable>::New();
   this->HistogramCache[image] = table.Get();
+  vtkSmartPointer<vtkImageData> const imageSP = image;
 
-  if (!this->Worker)
-    {
-    this->Worker = new HistogramWorker(this);
-    connect(this->Worker, SIGNAL(finished()), SLOT(histogramReady()));
-    }
-  else if (this->Worker->isRunning())
-    {
-    // FIXME: Queue, abort, something.
-    qDebug() << "Worker already running, skipping this one.";
-    return;
-    }
-  this->Worker->input = image;
-  this->Worker->output = table.Get();
-  this->Worker->start();
+  QMetaObject::invokeMethod(this->HistogramGen,
```

Comment:
Again here I don't see why you don't just call the C++ method rather than using invokeMethod